### ISSUE TITLE
test on ubfc-phys gives bad results.

### ISSUE THREAD

**Issue by ycyoon** [ubicomplab/rPPG-Toolbox#105]
I tested PURE, UBFC2, COFACE, and UBFC-Phys with mtts-can model.
Only a model trained with ubfc-phys gives bad results. (overfitted on the training dataset and fails on valid set)
I used s50~s56 as a valid set and trained with other videos.
Any idea? or does Anyone succeed in training with the UBFC-Phys dataset?

**Comment by yahskapar** [comment]
Did you test using this repo? What did your dataloader file look like (assuming you made a custom dataloader file for UBFC-PHYS)?

**Comment by ycyoon** [comment]
> Did you test using this repo? What did your dataloader file look like (assuming you made a custom dataloader file for UBFC-PHYS)?
> 
> EDIT: I should clarify that I'm asking if you used this repo since I see that you've posed the same question on the pyVHR repo.

I used this repo using a custom dataloader for UBFC-PHYS (with Normalized','Standardized' preprocessing). And as you said, I also used pyVHR repo and SRF-RPM(https://github.com/Dylan-H-Wang/SLF-RPM). The UBFC-PHYS dataset has been preprocessed in each repo's preprocessing instructions. However, I am failing to train with UBFC-Phys on all repo. I double-checked the UBFC-PHYS video and aligned the bvp signals and frames extracted from the video. I want to know if I'm missing something

**Comment by yahskapar** [comment]
Two things:

1) In the context of this repo, have you double-checked the sampling rate settings in your configuration file (.yaml)? It should be 35 Hz for UBFC-PHYS, instead of 30 Hz (for example, as it currently is for UBFC-rPPG). 

2) What does your preprocess_dataset_subprocess() function look like in your custom dataloader? After you acquire the blood volume pulse signal there, be sure to resample it, for example:`bvps = sample(bvps, frames.shape[0])`. This will ultimately match the length of your ground truth file with the length of your video, in this case using the sample function in utils/utils.py. 

May I also ask how exactly you are failing to train with UBFC-PHYS? Are you unable to even load the files for training, or are you somehow getting absurd losses? I was under the impression you simply got poor results based on your original post, but "failing to train" sounds much more extreme.
